internal/pkg/action/operator_uninstall.go

```diff
@@ -140,43 +133,29 @@ func (u *OperatorUninstall) deleteObjects(ctx context.Context, objs ...controlle
 	return waitForDeletion(ctx, u.config.Client, objs...)
 }
 
-func (u *OperatorUninstall) getInstallPlanResources(ctx context.Context, installPlanKey types.NamespacedName) (crds, csvs, others []controllerutil.Object, err error) {
-	installPlan := &v1alpha1.InstallPlan{}
-	if err := u.config.Client.Get(ctx, installPlanKey, installPlan); err != nil {
-		return nil, nil, nil, fmt.Errorf("get install plan: %v", err)
+func (u *OperatorUninstall) getObjectsToDelete(ctx context.Context, key types.NamespacedName) (csvs, crds []controllerutil.Object, err error) {
+	installedCSV := &v1alpha1.ClusterServiceVersion{}
+	if err = u.config.Client.Get(ctx, key, installedCSV); err != nil {
+		return
 	}
 
-	for _, step := range installPlan.Status.Plan {
-		obj := &unstructured.Unstructured{}
+	csvs = append(csvs, installedCSV)
```

Comment:
Two nits:
1. Now that we're guaranteed to have only one CSV, it might be cleaner to put the "get CSV" functionality into a separate `func (u *OperatorUninstall) getInstalledCSV(ctx, subscription)` function, and call that directly from `Run()`. Then we'd pass just the one csv to `deleteObjects()`.
2. We can make this function responsible for just returning the CRDs from the CSV, so it could be `func getCRDs(csv)`

And one suggestion:
1. When we get the CSV, we need to explicitly set the GVK with  `csv.SetGroupVersionKind()` so that the log output when deleting the CSV doesn't have an empty value for the kind being deleted.